deep_quoridor/src/quoridor.py

```diff
@@ -49,35 +50,49 @@ def is_valid_position_type(position: Position) -> bool:
 
 
 class Board:
-    """
-    Grid which has both walls and possible pawn positions.
-
-    Pawn positions are on even rows and columns, walls are on odd rows and columns. We use negative
-    values to represent free cells and walls, 0 for player 1, and 1 for player 2.
-    integers.
-    """
-
     # Possible values for each cell in the grid.
     OUT_OF_BOUNDS = -2  # Returned if a move is off of the board.
     FREE = -1
     # The player numbers start at 0 so that they can also be used as indices into the player positions array.
     PLAYER1 = 0
     PLAYER2 = 1
-    # Walls are stored as 10 + player number.We leave 2-9 free for implementing more than 2 players in the future.
-    PLAYER1_WALL = 10
-    PLAYER2_WALL = 11
+    WALL = 10
 
     def __init__(self, board_size: int = 9, max_walls: int = 10):
         self.board_size = board_size
         self.max_walls = max_walls
 
-        self._grid = np.full((board_size * 2 - 1, board_size * 2 - 1), Board.FREE, dtype=np.int8)
+        # We represent the board as a grid of cells alternating between wall cells and odd rows are player cells.
+        # To make some checks easier, we add a double border of walls around the grid.
+        self._grid = np.full((board_size * 2 + 3, board_size * 2 + 3), Board.FREE, dtype=np.int8)
+        self._grid[:2, :] = Board.WALL
+        self._grid[-2:, :] = Board.WALL
+        self._grid[:, :2] = Board.WALL
+        self._grid[:, -2:] = Board.WALL
+
+        # We also keep track of the walls in a (row x col x orientation) array so that we can give that
+        # representation to agents that expect it.
+        self._old_style_walls = np.zeros((self.board_size - 1, self.board_size - 1, 2), dtype=np.int8)
+
         self._players = [Player.ONE, Player.TWO]
         self._player_positions = [np.array([0, board_size // 2]), np.array([board_size - 1, board_size // 2])]
         for player, position in zip(self._players, self._player_positions):
-            self._grid[*position * 2] = player
+            self._grid[*(position * 2 + 2)] = player
```

Comment:
I'm not a fan of position being an np array.  I know it lets you do some math, but I'd prefer to use a simple int,int tupe and use helper functions when needed, e.g. here it could be `self._set_player_cell(position, player)`.

Reply:
I agree - since it would involve lots of little changes across `quoridor.py` I was avoiding including that change in this PR, but it's probably worth it. Do you like a tuple more than a `Position` class with row and col members?

Reply:
I think the easiest would be just a simple tuple (r,c) (i.e `tuple[int, int]`).

Reply:
A couple downsides with that:
- MoveAction, WallAction, Player are all classes. feels a bit odd for position to be treated differently
- can't do vector math, e.g. `position + offset`
- can't modify the `__str__(self)` function to pretty-print positions

But I'm not against it. I do think that change is outside of the scope of this PR though.

Reply:
The thing with using a class is that instead of just use for example `(1,2)` you need to use `Position(1,2)`. But it's nice to be able to have functions for offset or __str__.  We can discuss this more after this PR

Reply:
ok i actually did go ahead and change from array to tuple. We can still use numpy functions, but have to be a bit careful because they change the type. For example `np.subtract(x, y)` returns an ndarray even if x and y are tuples.